lib/interpreters/compute_several_solutions.rb

```diff
@@ -238,37 +238,47 @@ def self.find_best_heuristic(service_vrp)
     end
 
     def self.select_best_heuristic(vrp)
-      vehicles = vrp[:vehicles] || []
-      services = vrp[:services] || []
-      shipments = vrp[:shipments] || []
+      vehicles = vrp.vehicles
+      services = vrp.services
 
       loop_route = vehicles.any?{ |vehicle|
-        if (vehicle.start_point_id.nil? || vehicle.end_point_id.nil?) && (vehicle.start_point.nil? || vehicle.end_point.nil?)
+        if (vehicle.start_point_id.nil? || vehicle.end_point_id.nil?) &&
+           (vehicle.start_point.nil? || vehicle.end_point.nil?)
           true
         else
           start_point = vehicle.start_point
           end_point = vehicle.end_point
 
           vehicle.start_point_id == vehicle.end_point_id ||
-            start_point[:location] && end_point[:location] && start_point[:location][:lat] == end_point[:location][:lat] && start_point[:location][:lon] == end_point[:location][:lon] ||
-            vrp[:matrices] && start_point[:matrix_index] && end_point[:matrix_index] && vrp[:matrices].all?{ |matrix| matrix[:time] && matrix[:time][start_point[:matrix_index]][end_point[:matrix_index]] == 0 }
+            start_point.same_location?(end_point) ||
+            vrp.matrices && start_point.matrix_index && end_point.matrix_index &&
+              vrp.matrices.all?{ |matrix|
+                matrix.time && matrix.time[start_point.matrix_index][end_point.matrix_index] == 0
+              }
         end
       }
-      size_mtws = services.select{ |service| service[:timewindows].to_a.size > 1 }.size
-      size_rest = vehicles.collect{ |vehicle| vehicle[:rests].to_a.size }.sum
-      unique_configuration = vehicles.collect{ |vehicle| vehicle[:router_mode] }.uniq.size == 1 && vehicles.collect{ |vehicle| vehicle[:router_dimension] }.uniq.size == 1 &&
-                             vehicles.collect{ |vehicle| vehicle[:start_point_id] }.uniq.size == 1 && vehicles.collect{ |vehicle| vehicle[:end_point_id] }.uniq.size == 1
-
-      if vehicles.any?{ |vehicle| vehicle[:overall_duration] }
+      size_mtws = services.count{ |service| service.activity.timewindows.size > 1 }
+      size_rest = vehicles.map{ |vehicle| vehicle.rests.size }.sum
+      unique_configuration = vehicles.map(&:router_mode).uniq.size == 1 &&
+                             vehicles.map(&:router_dimension).uniq.size == 1 &&
+                             vehicles.map{ |vehicle| vehicle.start_point.id }.uniq.size == 1 &&
+                             vehicles.map{ |vehicle| vehicle.end_point.id }.uniq.size == 1
```

Comment:
no need for collect `map().uniq.size` -> `uniq().size`

@braktar you replaced `vehicle[:start_point_id]` with `vehicle.start_point.id` but the latter would fail if the vehicle have no start point, right? I suggest just replacing it with `vehicle.start_point_id` as follows:

```suggestion
      unique_configuration = vehicles.uniq(&:router_mode).size == 1 &&
                             vehicles.uniq(&:router_dimension).size == 1 &&
                             vehicles.uniq(&:start_point_id).size == 1 &&
                             vehicles.uniq(&:end_point_id).size == 1
```

Reply:
I think we should rather use `vehicle.start_point&.id` because every field with form `vehicle[:start_point_id]` corresponds to raw data provided by user. If `vrp.create` modifies this field in any way we will miss it. We should avoid using this form.

Reply:
I agree with @fonsecadeline `start_point_id` and `end_point_id` did not belongs to the model, they are co-products of our model creation from JSON or CSV. Any internal vehicle creation may not guarantee the presence of these fields.